src/OrleansBondUtils/BondSerializer.cs

```diff
@@ -191,15 +197,35 @@ private void Register(Type type)
             var clonerType = typeof(Cloner<>);
             var realType = clonerType.MakeGenericType(type);
             var clonerInstance = Activator.CreateInstance(realType);
-            var cloneMethod = realType.GetMethod("Clone").MakeGenericMethod(type);
-            var copierDelegate = cloneMethod.CreateDelegate(
-                    typeof(Func<,>).MakeGenericType(new[] { type, type }),
-                    clonerInstance);
             var serializer = new BondTypeSerializer(type);
             var deserializer = new BondTypeDeserializer(type);
-            CopierDictionary.TryAdd(type.TypeHandle, copierDelegate);
+            var sourceParameter = Expression.Parameter(typeof(object));
+            var instanceParameter = Expression.Parameter(typeof(object));
+            var method = typeof(BondSerializer).GetMethod("DeepCopy", BindingFlags.NonPublic | BindingFlags.Static).MakeGenericMethod(type);
+            var lambda = Expression.Lambda(
+                   typeof(Func<object, object, object>),
+                   Expression.Call(
+                       method,
+                       Expression.Convert(sourceParameter, type),
+                       instanceParameter),
+                   sourceParameter,
+                   instanceParameter);
+            var copierDelegate = (Func<object, object, object>)lambda.Compile();
+            ClonerInfoDictionary.TryAdd(type.TypeHandle, new ClonerInfo { Instance = clonerInstance, Delegate = copierDelegate } );
             SerializerDictionary.TryAdd(type.TypeHandle, serializer);
             DeserializerDictionary.TryAdd(type.TypeHandle, deserializer);
         }
+
+        private class ClonerInfo
+        {
+            public object Instance { get; set; }
+
+            public Func<object, object, object> Delegate { get; set; }
```

Comment:
Do we actually need properties for these? Otherwise, just have private readonly fields and initialize in a constructor